model/godleyTableWindow.cc

```diff
@@ -846,42 +847,49 @@ namespace {
 	if (clickType(x,y)==colWidget) {
 	    unsigned visibleCol=c-scrollColStart+1;
         if (c<colWidgets.size() && visibleCol < colLeftMargin.size()) {
-		    auto& moveVar=godleyIcon->table.cell(0,c);		
+		    auto moveVar=godleyIcon->table.getCell(0,c);		
 		    auto oldAssetClass=godleyIcon->table._assetClass(c);
-            if (colWidgets[c].button(x-colLeftMargin[visibleCol])==3) {
-				auto targetAssetClass=godleyIcon->table._assetClass(c+1);
-				if (targetAssetClass!=oldAssetClass && !moveVar.empty())
-				    tmpStr=constructMessage(targetAssetClass,oldAssetClass,moveVar);
-			}
-			else if (colWidgets[c].button(x-colLeftMargin[visibleCol])==2 && oldAssetClass==1 && godleyIcon->table._assetClass(c-1)!=1) {
-				auto targetAssetClass=godleyIcon->table._assetClass(c+1);
-				if (targetAssetClass!=oldAssetClass && !moveVar.empty())
-				    tmpStr=constructMessage(targetAssetClass,oldAssetClass,moveVar);
-			}
-			else if (colWidgets[c].button(x-colLeftMargin[visibleCol])==2) {
-				auto targetAssetClass=godleyIcon->table._assetClass(c-1);
-				if (targetAssetClass!=oldAssetClass && !moveVar.empty())
-				    tmpStr=constructMessage(targetAssetClass,oldAssetClass,moveVar);
-			}
+		    auto targetAssetClassPlus=godleyIcon->table._assetClass(c+1);
+		    auto targetAssetClassMinus=godleyIcon->table._assetClass(c-1);
+            if (colWidgets[c].button(x-colLeftMargin[visibleCol])==3 && oldAssetClass!=3) {
+		    	if (targetAssetClassPlus!=oldAssetClass && !moveVar.empty() && targetAssetClassPlus!=3 && targetAssetClassPlus!=0)
+		    	    tmpStr=constructMessage(targetAssetClassPlus,oldAssetClass,moveVar);
+		    	else if ((targetAssetClassPlus==3 || targetAssetClassPlus==0) && !moveVar.empty())
+		    	    tmpStr="Cannot convert stock variable to an equity class";    
+		    }
+		    else if (colWidgets[c].button(x-colLeftMargin[visibleCol])==2 && oldAssetClass==1 && oldAssetClass!=3 && targetAssetClassMinus!=1) {
+		    	if (targetAssetClassPlus!=oldAssetClass && !moveVar.empty() && targetAssetClassPlus!=3 && targetAssetClassPlus!=0)
+		    	    tmpStr=constructMessage(targetAssetClassPlus,oldAssetClass,moveVar);
+		    	else if ((targetAssetClassPlus==3 || targetAssetClassPlus==0) && !moveVar.empty())
+		    	    tmpStr="Cannot convert stock variable to an equity class"; 		    	    
+		    }
+		    else if (colWidgets[c].button(x-colLeftMargin[visibleCol])==2 && oldAssetClass!=3) {
```

Comment:
Better to use symbolic values when comparing enums, ie oldAssetClass!=AssetClass::equity